# People on Banknotes

Whose faces appear on banknotes?

The file `people-on-banknotes.csv` contains data about individuals featured on banknotes from 38 countries. This dataset spans all 22 subregions and sub-subregions of the world, as defined by the United Nations Statistics Division's geoscheme.

It profiles 241 people, detailing their occupations and the year they first appeared on a banknote. Additionally, it includes their year of death — or `NaN` if they were still alive when the dataset was compiled.

Most banknotes were issued after the featured individual’s death. The column `first_death_diff` calculates the difference between the year of their first appearance on a banknote and their year of death (or remains `NaN` if the person was still living at the time of curation).




In [1]:
# FOR GOOGLE COLAB ONLY.
# Uncomment and run the code below. A dialog will appear to upload files.
# Upload 'people-on-banknotes.csv'.

# from google.colab import files
# uploaded = files.upload()

In [2]:
import pandas as pd

df = pd.read_csv('people-on-banknotes.csv')
df

,country,currency,name,gender,occupation,value,first_appearance,death,first_death_diff,currency_code
0,Argentina,Argentine Peso,Eva Perón,F,Activist,100,2012,1952,60.0,ARS
1,Argentina,Argentine Peso,Julio Argentino Roca,M,Head of Gov't,100,1988,1914,74.0,ARS
2,Argentina,Argentine Peso,Domingo Faustino Sarmiento,M,Head of Gov't,50,1999,1888,111.0,ARS
3,Argentina,Argentine Peso,Juan Manuel de Rosas,M,Politician,20,1992,1877,115.0,ARS
4,Argentina,Argentine Peso,Manuel Belgrano,M,Founder,10,1970,1820,150.0,ARS
...,...,...,...,...,...,...,...,...,...,...
274,Venezuela,Venezuelan Bolivar,Francisco de Miranda,M,Military,200,1968,1816,152.0,VES
275,Venezuela,Venezuelan Bolivar,Simón Rodrigues,M,Educator,20,2007,1854,153.0,VES
276,Venezuela,Venezuelan Bolivar,Ezequiel Zamora,M,Military,100,2018,1860,158.0,VES
277,Venezuela,Venezuelan Bolivar,Rafael Urdaneta,M,Head of Gov't,10,2018,1845,173.0,VES


### Quick cleaning

The same person can appear on multiple banknotes. Below we drop the `value` column and remove duplicate people.

In [3]:
df = df.drop(columns=['value'])
df = df.drop_duplicates(subset="name")
df

,country,currency,name,gender,occupation,first_appearance,death,first_death_diff,currency_code
0,Argentina,Argentine Peso,Eva Perón,F,Activist,2012,1952,60.0,ARS
1,Argentina,Argentine Peso,Julio Argentino Roca,M,Head of Gov't,1988,1914,74.0,ARS
2,Argentina,Argentine Peso,Domingo Faustino Sarmiento,M,Head of Gov't,1999,1888,111.0,ARS
3,Argentina,Argentine Peso,Juan Manuel de Rosas,M,Politician,1992,1877,115.0,ARS
4,Argentina,Argentine Peso,Manuel Belgrano,M,Founder,1970,1820,150.0,ARS
...,...,...,...,...,...,...,...,...,...
274,Venezuela,Venezuelan Bolivar,Francisco de Miranda,M,Military,1968,1816,152.0,VES
275,Venezuela,Venezuelan Bolivar,Simón Rodrigues,M,Educator,2007,1854,153.0,VES
276,Venezuela,Venezuelan Bolivar,Ezequiel Zamora,M,Military,2018,1860,158.0,VES
277,Venezuela,Venezuelan Bolivar,Rafael Urdaneta,M,Head of Gov't,2018,1845,173.0,VES


### Project Ideas

- What proportion of individuals featured are male versus female?
	- Hint: Use `value_counts(normalize=True)` to calculate percentages.

- Are writers or politicians more commonly depicted?

- What percentage of featured individuals are musicians?

- What percentage of banknotes were issued before the person’s death?
	- Hint: Look for negative values or NaN in `first_death_diff`.

- Who is the oldest historical figure in the dataset?

- Which countries feature the oldest historical figures on their banknotes?
	- Hint: Group by country and aggregate the year of death using the median. Sort the results.

- What percentage of individuals died at least 100 years before appearing on a banknote?

- Which individuals appeared on a banknote just one year after their death?


Question 1: 
What proportion of individuals featured are male versus female?
	- Hint: Use `value_counts(normalize=True)` to calculate percentages.

In [14]:
gen_prop = df['gender'].value_counts(normalize=True) * 100#returns the proportion instead of raw count
gen_prop

gender
M    78.008299
F    21.991701
Name: proportion, dtype: float64

**Answer:** Approximately 78% of the featured persons are male, and around 21.99% are female.

Question 2:
Are writers or politicians more commonly depicted?

In [16]:
occu_prop = (df['occupation'].value_counts(normalize=True) * 100).sort_values(ascending= False)
occu_prop

occupation
Writer              18.672199
Head of Gov't       15.767635
Politician          11.203320
Founder             10.373444
Revolutionary        8.713693
STEM                 6.639004
Military             5.394191
Visual Artist        5.394191
Musician             4.979253
Monarch              4.979253
Other                2.074689
Activist             1.659751
Educator             1.659751
Religious figure     1.244813
Performer            1.244813
Name: proportion, dtype: float64

**Answer:** Yes, writers are the most commonly depicted, while politicaians ranks third.

Question 3:
What percentage of featured individuals are musicians?

**Answer:** Approximately 4.98% of the featured individuals are musicians.

Question 4: 
What percentage of banknotes were issued before the person’s death?
	- Hint: Look for negative values or NaN in `first_death_diff`.

In [25]:
bef_death = df.query('first_death_diff < 0')
bef_death

,country,currency,name,gender,occupation,first_appearance,death,first_death_diff,currency_code
14,Bangladesh,Bangladeshi Taka,Bangabandhu Sheikh Mujibur Rahman,M,Founder,1972,1975,-3.0,BDT
119,Indonesia,Indonesian Rupiah,Soekarno,M,Head of Gov't,1945,1970,-25.0,IDR
135,Jamaica,Jamaican Dollar,Michael Norman Manley,M,Head of Gov't,1970,1992,-22.0,JMD
150,Malawi,Malawian Kwacha,Dr. Hastings Kamuzu Banda,M,Head of Gov't,1971,1997,-26.0,MWK
151,Malawi,Malawian Kwacha,Rose Lomathinda Chibambo,F,Politician,2012,2016,-4.0,MWK
168,New Zealand,New Zealand Dollar,Edmund Hillary,M,Other,1992,2008,-16.0,NZD
182,Papua New Guinea,Papua New Guinean Kina,Michael Somare,M,Founder,1989,2021,-32.0,PGK
216,South Africa,South African Rand,Nelson Mandela,M,Head of Gov't,2012,2013,-1.0,ZAR
235,Turkey,Turkish Lira,Kemal Atatürk,M,Founder,1927,1938,-11.0,TRY


Question 5: Who is the oldest historical figure in the dataset?

Question 6: Which countries feature the oldest historical figures on their banknotes?
	- Hint: Group by country and aggregate the year of death using the median. Sort the results.

Question 7: What percentage of individuals died at least 100 years before appearing on a banknote?

Question 8: Which individuals appeared on a banknote just one year after their death?

In [4]:
# YOUR CODE HERE (add additional cells as needed)